### PR TITLE
Domains: Replace claim domain notice with UpsellNudge

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

This updates the `Notice` component to an `UpsellNudge` block following along with the work in #38778

Some notes:
* Consider this more a "try" than a formal change. :)
* This makes for two primary calls to action one after the other, which is confusing/looks weird, and it doesn't make sense to me to display this notice when there's a prominent nudge in the sidebar. I'm guessing the reasoning is because people would add yet *another* domain instead of claiming the free one first?
* Maybe we roll this notice into @jancavan 's notices work, rather than making it an upsell?

**Before**

Screen Shot 2020-04-24 at 2 07 38 PM

**After**

Screen Shot 2020-04-24 at 2 07 51 PM

#### Testing instructions

* Switch to this PR on a site with a Premium or Business plan that does not yet have a domain attached
* Go to Manage -> Domains
* You should see the upsell at the top of the screen. Tracks events (impression and click) should remain the same.

### CODE REVIEW

**Comment by sixhours** [comment]
I am now realizing that it's the `DomainToPlanNudge` that should be updated. 🙃 

That said, we replaced a similar sidebar `Notice` with `UpsellNudge` so I'll leave this open for discussion.

**Comment by jancavan** [comment]
I'm wondering if we could surface the "Best Match" domain here with a "Claim" CTA next to it, then there would be a secondary button if they wish to look for something else (this button would take them to Domain Search.)

This is obviously out of scope, but just wanted to share this idea:

Screen Shot 2020-04-24 at 12 00 19 PM

**Comment by jancavan** [comment]
> I'm guessing the reasoning is because people would add yet another domain instead of claiming the free one first?

I'm thinking maybe because the sidebar isn't visible on mobile at all times? So people would still see this nudge when on the Domains screen.

That said, what if for now we just get rid of this nudge on desktop since we already have the sidebar visible as well? But we would show it on mobile. We'd probably have to look at the numbers though.

**Comment by sixhours** [comment]
> I'm thinking maybe because the sidebar isn't visible on mobile at all times? So people would still see this nudge when on the Domains screen.

Oh, that's a really good point I hadn't considered.  Curious to get the backstory from @Automattic/cobalt , if any :)

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 1]
```diff
@@ -110,24 +108,19 @@ export class List extends React.Component {
 		const { translate } = this.props;
 
 		return (
-			<Notice
-				status="is-success"
-				showDismiss={ false }
-				text={ translate( 'Free domain available' ) }
-				icon="info-outline"
+			<UpsellNudge
+				callToAction={ translate( 'Claim Free Domain' ) }
```
Comment (kwight) [inline client/my-sites/domains/domain-management/list/index.jsx:112]: Are these button texts supposed to be sentence-case?

**Comment by kwight** [comment]
Works great 👍 

> I'm guessing the reasoning is because people would add yet another domain instead of claiming the free one first?

Ha, right, yes, excellent point. As @sixhours said, still awkward to have three total primary buttons 🙃

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 2]
```diff
@@ -110,24 +108,19 @@ export class List extends React.Component {
...
-				>
-					{ translate( 'Claim Free Domain' ) }
-					<TrackComponentView
-						eventName={ 'calypso_domain_credit_reminder_impression' }
-						eventProperties={ { cta_name: 'domain_info_notice' } }
-					/>
-				</NoticeAction>
-			</Notice>
+				forceDisplay
+				href={ `/domains/add/${ this.props.selectedSite.slug }` }
```
Comment (kwight) [inline client/my-sites/domains/domain-management/list/index.jsx:116]: It's odd to me to see some props destructured (`translate`) but not others, although it's easy to argue `hasDomainCredit` and `selectedSite` aren't used more than once – whichever you think is best 🙃

Comment (sixhours) [reply client/my-sites/domains/domain-management/list/index.jsx:116]: It looks like `{ hasDomainCredit }` is already declared as a selector, so it throws an error when you destructure it...it also looks like the function returns null when `hasDomainCredit` is false, so it never gets to the other two. I destructured `selectedSite` for consistency with `translate`, though.

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 3]
```diff
@@ -107,27 +105,22 @@ export class List extends React.Component {
...
+		const { translate, selectedSite } = this.props;
 
 		return (
-			<Notice
-				status="is-success"
-				showDismiss={ false }
-				text={ translate( 'Free domain available' ) }
-				icon="info-outline"
+			<UpsellNudge
+				callToAction={ translate( 'Claim free domain' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/list/index.jsx:112]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **27** times:
`translate( 'Claim Free Domain' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-claim-notice)

File: client/my-sites/domains/domain-management/list/index.jsx (hunk 1 above)
Comment (sixhours) [reply client/my-sites/domains/domain-management/list/index.jsx]: Ayuh :)

**Comment by klimeryk** [comment]
> Oh, that's a really good point I hadn't considered. Curious to get the backstory from @Automattic/cobalt , if any :)

No story that I know of :) I'm afraid we were flying without any long-term dedicated designers for 2-3 years. And the result is often subpar UX. I'd love to reverse that trend but we need to do that in a more systematic and general approach, instead of focusing on small changes and micro-optimizations.

Should we show this upsell at all? Is it converting users? Is the copy explaining clearly what's available? Looking at it now with a fresh perspective, it's actually a bit misleading - it's a free domain, but only for one year. This also ties into the discussion that @wensco often brings up (with very good reasons) about how the domain credit is confusing and it might make sense from user perspective for it to be a one time coupon instead. "You bought a plan, you get this one time coupon you can redeem for one year of a custom domain". I'd imagine that's easier to understand than a "domain credit" for a free domain. It probably made more sense when the credit ensured the domain would be renewed for free as long as the user had a paid plan, but that's no more since over a year ago.

I'd love to have these conversations and figure out a way forward, but I'm not sure how and where. I'm seeing small "fix the flow" fixes here and there, but I'm afraid we're focusing on the wrong things here. If there's dedicated time for designers to fix issues like this, I'd love it if we could channel it to closer collaboration with the product team (in this case @Automattic/cobalt) and Happiness to figure out where the biggest pain points are.

**Comment by gwwar** [comment]
@sixhours what's still needed for this one?

**Comment by sixhours** [comment]
> @sixhours what's still needed for this one?

I'm going to scrap this since it's not an upsell and I misunderstood the original goal (which was to convert `DomainToPlanNudge`, not the notice, done in #41453).

**Pull request closed without merging**